### DISCUSSION TITLE
Validate all fields on a field update

### DISCUSSION THREAD

**Discussion by Ansalibrahim** [react-hook-form/react-hook-form#7356] [RFC]
I am trying to implement a form with a field array, the array does have multiple fields. I need to check whether a field value repeats. Unfortunately, only current focus field validation is triggered when the data is updated. I have to show the error like "value is repeated" in all the fields. 

Is there a way to do this with react-hook-form?

**Comment by Ansalibrahim** [comment]
Here is a minimalistic sample if you want to check https://codesandbox.io/s/react-hook-form-usefieldarray-forked-uwuu9?file=/src/index.js 

In the example, the append will add a new row. change the first name in the second row to Bill. while typing the field will be validated and shows the error "Repeated". 

I need to show the error on every row which is repeated. In this case, I need to show the error "Repeated" in both rows.

**Comment by bluebill1049** [comment]
`onChange` with `trigger()`?

**Reply by Ansalibrahim** [reply]
It seems like `trigger` has no impact.

**Reply by Ansalibrahim** [reply]
The issue is resolved by using `trigger()`. The sample app I created doesn't work with `trigger()` for some reason.

Thank you @bluebill1049 for the help and timely response :-)

**Reply by MosheSchnitzler** [reply]
@Ansalibrahim I believe that you need to pass in the full path to the field in trigger(), for example:

```
interface User {
  basicInfo: BasicInfo;
  paymentInfo: Payment;
}

const form = useForm<User>({ values: props.user })

// validate a field
await form.trigger('basicInfo.email');
```
